src/Stratis.FederatedPeg.Features.FederationGateway/SourceChain/MaturedBlocksProvider.cs

```diff
@@ -50,6 +52,7 @@ private async Task<List<ChainedHeader>> GetChainedHeadersAsync(int blockHeight,
                 blockHashes.Add(chainedHeader.HashBlock);
             }
 
+            // // TODO: this should be called from consensus manager.GetBlockData().
```

Comment:
code in this method gets headers from given block height till the tip. But most likely some of the blocks won't be pushed to repo yet, just to BSQ. So for sure this TODO should be resolved, it's a bug.